### PR TITLE
feat: add example users api to test mongodb integration

### LINKED ISSUE #13: Setup Mongodb Atlas Cluster

### PR DESCRIPTION
**Pull request by alexnaish**
Closes #13

### CODE REVIEW

File: pages/api/users.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import nc from 'next-connect';
+
+import User from '../../models/User';
+import { connectionMiddleware } from '../../utils/mongo';
+
+const cleanseVariable = (param) => {
+    return decodeURIComponent(param.toString()).replace(/([^a-z0-9_\-.%@ ]+)/gi, '-');
```
Comment (samnaish) [inline pages/api/users.js:7]: What is happening with this regex is it going to look through the email string and replace with "-"?
wouldn't replaceAll() be better?

**Comment by alexnaish** [comment]
This does do a replace all, the “gi” part on the regex means “global” and “case insensitive”. It’s effectively turning any character not in that list into a dash (stopping any shenanigans)

File: package.json [hunk 2]
```diff
@@ -25,7 +25,9 @@
     },
     "homepage": "https://github.com/AlexNaishLtd/management-portal#readme",
     "dependencies": {
+        "mongoose": "^5.12.2",
         "next": "^10.0.9",
+        "next-connect": "^0.10.1",
```
Comment (samnaish) [inline package.json:30]: What does this do?
Does it allow/make things easier to use with next.js

**Review by samnaish — APPROVED** [review]

**Comment by alexnaish** [comment]
The documentation is here: https://www.npmjs.com/package/next-connect

It essentially lets us easily use middleware (like with express) before our
main functions execute and also lets us handle different HTTP methods
better. For instance, for the /users endpoint - we would want GET to list
all users but a POST request to create a new user and for BOTH we would
want to be able to load the mongodb code beforehand. Does that make sense?

**Pull request merged**